<launch>
	<!-- Arguments -->
	<arg name="Field" default="none"/>
	<arg name="map_file" default="$(find Four-wheel_omni)/map/map_world_CG.yaml"/>
	<arg name="open_rviz" default="true"/>

	<!-- odom -->
	<group if="$(eval Field == 'Red')">
		<include file="$(find Four-wheel_omni)/launch/odom_gazebo.launch">
			<arg name="map_file" value="$(find Four-wheel_omni)/map/map_world_Red.yaml" />
			<arg name="open_rviz" default="$(arg open_rviz)"/>
		</include>
	</group>
	<group if="$(eval Field == 'Blue')">
		<include file="$(find Four-wheel_omni)/launch/odom_gazebo.launch">
			<arg name="map_file" value="$(find Four-wheel_omni)/map/map_world_Blue.yaml" />
			<arg name="open_rviz" default="$(arg open_rviz)"/>
		</include>
	</group>
	<group if="$(eval Field == 'none')">
		<include file="$(find Four-wheel_omni)/launch/odom_gazebo.launch">
			<arg name="map_file" value="$(arg map_file)" />
			<arg name="open_rviz" default="$(arg open_rviz)"/>
		</include>
	</group>


	<!-- Path Planner-->
	<node pkg="Four-wheel_omni" type="PathPlanner.py" name="PathPlanner" output="screen"/>

	<!-- Path Follower -->
	<node pkg="Four-wheel_omni" type="PurePursuit.py" name="PurePursuit" output="screen"/>

	<node pkg="jsk_rviz_plugins"
			type="rosconsole_overlay_text.py"
			respawn="true"
			name="rosconsole_overlay_text">
		<param name="reverse_lines" value="true"/>
	</node>
</launch>
